mxtools/scans.py

```diff
@@ -86,3 +94,65 @@ def setup_vector_program(num_images, angle_start, angle_end, exposure_period_per
         vector.hold,
         0,
     )
+
+
+def setup_eiger_exposure(eiger, exposure_time, exposure_period):
+    yield from bps.mv(eiger.acquire_time(exposure_time))
+    yield from bps.mv(eiger.acquire_period(exposure_period))
+
+
+def setup_eiger_triggers(eiger, mode, num_triggers, exposure_per_image):
+    yield from bps.mv(eiger.trigger_mode, mode)
+    yield from bps.mv(eiger.num_triggers, num_triggers)
+    yield from bps.mv(eiger.trigger_exposure, exposure_per_image)
+
+
+def setup_eiger_arming(
+    eiger,
+    start,
+    width,
+    num_images,
+    exposure_per_image,
+    file_prefix,
+    data_directory_name,
+    file_number_start,
+    x_beam,
+    y_beam,
+    wavelength,
+    det_distance_m,
+):
+    yield from bps.mv(eiger.save_files, 1)
+    yield from bps.mv(eiger.file_owner, getpass.getuser())
+    yield from bps.mv(eiger.file_owner_grp, grp.getgrgid(os.getgid())[0])
+    yield from bps.mv(eiger.file_perms, 420)
+    file_prefix_minus_directory = str(file_prefix)
+    try:
+        file_prefix_minus_directory = file_prefix_minus_directory[
+            file_prefix_minus_directory.rindex("/") + 1 : len(file_prefix_minus_directory)  # noqa: E203
+        ]
+    except ValueError:
+        pass
+
+    yield from bps.mv(eiger.acquire_time, exposure_per_image)
+    yield from bps.mv(eiger.acquire_period, exposure_per_image)
+    yield from bps.mv(eiger.num_images, num_images)
+    yield from bps.mv(eiger.file_path, data_directory_name)
+    yield from bps.mv(eiger.fw_name_pattern, f"{file_prefix_minus_directory}_$id")
+    yield from bps.mv(eiger.sequence_id, file_number_start)
```

Comment:
I am not sure if those 3 parameters will be overwritten by `eiger.stage()`... I guess we need to create our custom method to overwrite https://github.com/bluesky/ophyd/blob/d0bc07ef85cf2d69319a059485a21a5cdbd0e677/ophyd/areadetector/filestore_mixins.py#L403-L422 (uuid4-based file names).

Reply:
after our conversation, I agree with what you are proposing. we agree to implement this custom method in a later PR